File: represent/bench/c_flat/FormalBench_0239__FormalBenchHF__diverse__rule_1__IsNumDecagonal_llm_1.c
#include <stdlib.h>

int isNumDecagonal(int num);

int isNumDecagonal(int num) {

		int *arr = (int *)malloc(sizeof(int) * (num + 1));
int arr_len = num + 1;
		arr[0] = 0;
		for (int index = 0; index <= num - 1; index++) {
			arr[index + 1] = arr[index] + 8 * index + 1;
		}
		return arr[num];
}

File: represent/bench/java_flat/FormalBench_0239__FormalBenchHF__diverse__rule_1__IsNumDecagonal_llm_1.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class IsNumDecagonal {

	public static int isNumDecagonal(int num) {
		int[] arr = new int[num + 1];
		arr[0] = 0;
		for (int index = 0; index <= num - 1; index++) {
			arr[index + 1] = arr[index] + 8 * index + 1;
		}
		return arr[num];
	}
}
